resources/migrations/trash/h2.sql

```diff
@@ -0,0 +1,101 @@
+-- DASHBOARDS
+-- Set `trashed_directly`.
+UPDATE report_dashboard
+SET trashed_directly = COALESCE(
+  -- If the collection is archived then it was *not* trashed directly
+  (SELECT NOT collection.archived FROM collection WHERE collection.id = report_dashboard.collection_id),
+  false
+  )
+WHERE archived = true;
```

Comment:
Yes, that's right. One thing to note is that we're doing a kind of "best effort" thing here - we can't really know for sure whether something was trashed directly or not. For example, maybe someone trashed a dashboard, then later on trashed the collection it used to be in. We'll mark it as `trashed_directly: false` when it wasn't.

I think that's ok/inevitable, but just wanted to call it out. The knowledge of what was `trashed_directly` has been lost because we didn't record it at the moment it happened. We just do the best we can to do something reasonable.